### ISSUE TITLE
ux_mode not an option parameter for gapi.auth2.init

### ISSUE THREAD

**Issue by monisha3107** [google/google-api-javascript-client#352]
I am trying to avoid a browser pop-up and use ux_mode='redirect' during init() or signIn() and it seems like I cannot use this as one of the option parameters to either of those methods. Has this been removed/modified?

**Comment by TMSCH** [comment]
Could you post a snippet of code that reproduces the issue? We haven't changed anything related to that recently.

**Comment by monisha3107** [comment]
`gapi.auth2.init({client_id: "my_client_id", scope: "profile email", ux_mode: "redirect"})`

ux_mode isn't a valid parameter. I only have option of setting the following: 

client_id 
cookie_policy 
scope 
fetch_basic_profile
hosted_domain 
openid_realm

But your documentation here says otherwise. https://developers.google.com/identity/sign-in/web/reference#gapiauth2clientconfig

**Comment by TMSCH** [comment]
I tried just now and it works for me. How do you load the library? Could you post a full snippet that reproduces the issue?

**Comment by monisha3107** [comment]
**index.html** has this and loads **appcomponent.html**: 
`<script src="https://apis.google.com/js/platform.js" async defer></script>
`

 **appcomponent.html** has this:
```
<google-signin id="signin"
               style="display: inline-block;"
               [clientId]="clientId"
               [scope]="scope"
               (googleSignInSuccess)="onGoogleSignIn()">
</google-signin>

<button style="float:right; height: 36px; display: inline-block;" class="btn btn-default" (click)="onGoogleSignOut();">Sign out</button>
```

And `appcomponent.ts` has the following: 
 ```
 onGoogleSignIn() {

    let googleAuth = gapi.auth2.getAuthInstance();
    this.user = googleAuth.currentUser.get();

    if (googleAuth.currentUser.get().isSignedIn()) {
      googleAuth.signIn().then((data) => {
        this.token = data.Zi.access_token;
      });
    }
    else
    {
      let authResp = this.user.getAuthResponse();
      this.token = authResp.access_token;
    }
  }

  onGoogleSignOut() {
    gapi.auth2.getAuthInstance().signOut().then(function () {
      console.log('User signed out.');
    });
  }
```

This works just fine. But I wanted to not have the pop-up and instead have the user be redirected to another page to select the google account to login with. So, I was trying to specify _ux_mode_ in the signIn() parameters or add in a onLoad() function with init().

**Comment by TMSCH** [comment]
Have you tried adding it in the `signIn` call?

```
googleAuth.signIn({ux_mode: 'redirect'});
```

Also, you're using a component, `google-signin` that is not from the library. You should make sure that this supports ux_mode, as you probable have to pass it as a parameter.

**Comment by monisha3107** [comment]
Yes, I have tried that and it's same issue. Ok, if not `google-signin`, what component should I be using?

**Comment by TMSCH** [comment]
You can file a bug at https://github.com/GoogleWebComponents/google-signin to ask them to add support for `ux_mode`.

**Comment by dvanderb** [comment]
This looks like a question about Google Login and Authentication. Please address this question to the google-signin tag on stackoverflow(https://stackoverflow.com/questions/tagged/google-signin)

If you believe this is in error, please reopen the issue.
